api/openapi_server/repositories/service_provider_repository.py

```diff
@@ -54,14 +56,14 @@ def get_service_provider_by_id(self, provider_id: int) -> Optional[HousingProgra
         :type provider_id: int
         """
         with DataAccessLayer.session() as session:
-            return session.query(HousingProgramServiceProvider).get(provider_id)
+            return session.get(HousingProgramServiceProvider, provider_id)
             
     def get_service_providers(self) -> List[HousingProgramServiceProvider]:
         """
         Get a list of all housing program service providers.
         """
         with DataAccessLayer.session() as session:
-            return session.query(HousingProgramServiceProvider).all()
+            return session.execute(select(HousingProgramServiceProvider)).scalars().all()
```

Comment:
Why does SQLA have two ways of doing this `session.scalars(...).all()` and `session.execute(...).scalars().all()`

Reply:
The `session.execute()` is the more general, SQL like approach. This returns [Row](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.Row) objects (containing individual columns), which need to be unwrapped into the ORM objects (using `scalars()`).
`session.scalars()` skips the Row objects part and returns the ORM object directly.
Both are valid and equivalent, I just chose the one that was listed first in the translation table.

Is there a particular preference for one over another?

Reply:
https://docs.sqlalchemy.org/en/20/orm/queryguide/select.html#orm-queryguide-select-orm-entities

This portion of the doc talks about them both

Reply:
Yeah that portion of the docs says that `session.scalars(...)` is equivalent to `session.execute(...).scalars()`. It is just a shortcut if you need to call the `scalars()` method on the `Result` object returned by a session statement execution. 

With that in mind, you can consolidate those three lines of code into a single return statement:
```
return session.scalars(select(HousingProgramServiceProvider)).all()
```